src/test/scala/nl.knaw.dans.easy.validatebag/EasyValidateDansBagServletSpec.scala

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (C) 2018 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.validatebag
+
+import java.net.URI
+
+import org.apache.commons.configuration.PropertiesConfiguration
+import org.eclipse.jetty.http.HttpStatus.{ BAD_REQUEST_400, OK_200 }
+import org.scalatra.test.scalatest.ScalatraSuite
+
+class EasyValidateDansBagServletSpec extends TestSupportFixture with ServletFixture with ScalatraSuite  {
+  private val app = new EasyValidateDansBagApp(Configuration("0", createProperties(), Seq(new URI("http://creativecommons.org/licenses/by-sa/4.0"))))
+  private val validateBagServlet = new EasyValidateDansBagServlet(app)
+  addServlet(validateBagServlet, "/*")
+
+  "the validate handler" should "return a 200 and the response when presented a valid bag uri" in {
+    post(uri = s"/validate?infoPackageType=SIP&uri=file://${ bagsDir.path.toAbsolutePath }/valid-bag", headers = Seq(("Accept", "Application/Json"))) {
+      status shouldBe OK_200
+      body should include("Is compliant: true")
+    }
+  }
+
+  it should "return a 200 and a response including 'compliant: false' and reasons when presented an invalid bag" in {
+    post(uri = s"/validate?infoPackageType=SIP&uri=file://${ bagsDir.path.toAbsolutePath }/metadata-correct", headers = Seq(("Accept", "Application/Json"))) {
+      status shouldBe OK_200
+      body should include(" - [1.2.4(a)] bag-info.txt must contain exactly one 'Created' element; number found: 0")
+      body should include(" - [1.3.1(a)] Mandatory file 'manifest-sha1.txt' not found in bag.")
+      body should include("Is compliant: false")
+    }
+  }
+
+  it should "return a" +
+    " 400 if presented a non existing bag uri" in {     //TODO shouldn't this return a 404?
```

Comment:
@janvanmansum could you take a look at this TODO?

Reply:
I **don't** think it should be 404. The bag URL is a query parameter. A 404 would seem to indicate that **the base URL** (so without the query parameters) does not exist.